File: ecos-records/src/main/java/ru/citeck/ecos/records2/RecordsProperties.java
package ru.citeck.ecos.records2;

import lombok.Data;

import java.util.Map;

@Data
public class RecordsProperties {

    private String appName = "";
    private String appInstanceId = "";
    private RestProps rest;
    private Map<String, App> apps;
    private Map<String, String> sourceIdMapping;

    /**
     * Disable remote records for testing.
     */
    private boolean forceLocalMode;

    /**
     * Used by gateway.
     */
    private boolean gatewayMode;
    private String defaultApp;

    @Data
    public static class App {
        private String recBaseUrl;
        private String recUserBaseUrl;
        private Authentication auth;
    }

    @Data
    public static class RestProps {
        private Boolean secure;
    }

    @Data
    public static class Authentication {
        private String username;
        private String password;
    }
}

File: ecos-records/src/main/java/ru/citeck/ecos/records2/RecordsProperties.kt
package ru.citeck.ecos.records2

class RecordsProperties {

    var appName = ""
    var appInstanceId = ""
    var rest: RestProps = RestProps()
    var apps: Map<String, App> = emptyMap()
    var sourceIdMapping: Map<String, String> = emptyMap()

    /**
     * Disable remote records for testing.
     */
    var forceLocalMode = false

    /**
     * Used by gateway.
     */
    var gatewayMode = false
    var defaultApp: String = ""

    class App {
        var recBaseUrl: String? = null
        var recUserBaseUrl: String? = null
        var auth: Authentication? = null
    }

    class RestProps {
        var secure: Boolean = false
    }

    class Authentication {
        var username: String? = null
        var password: String? = null
    }
}
